### PR TITLE
Resource sets

### PR DESCRIPTION
**Pull request by julien-f**

### CODE REVIEW

File: src/api/resource-set.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+export function create ({ name }) {
+  return this.createResourceSet(name)
```
Comment (marsaud) [inline src/api/resource-set.js]: Could we handle optional arrays of subjects and objets, here ?

File: src/api/resource-set.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+export function addObject ({ object }) {
+  return this.addObjectToResourceSet(object)
+}
+
+addObject.permission = 'admin'
+
+// -------------------------------------------------------------------
+
+export function removeObject ({ object }) {
+  return this.removeObjectFromResourceSet(object)
```
Comment (marsaud) [inline src/api/resource-set.js]: An object, really, not an object id ?

Comment (marsaud) [inline src/api/resource-set.js]: Why not an array of objects ?

File: src/api/resource-set.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+removeObject.permission = 'admin'
+
+// -------------------------------------------------------------------
+
+export function addSubject ({ subject }) {
+  return this.addSubjectToResourceSet(subject)
```
Comment (marsaud) [inline src/api/resource-set.js]: Same questions than for objects

File: src/api/resource-set.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    ? this.getAllResourceSets()
+    : this.getResourceSet(id)
+}
+
+get.permission = 'admin'
+
+// -------------------------------------------------------------------
+
+export function addObject ({ object }) {
+  return this.addObjectToResourceSet(object)
```
Comment (marsaud) [inline src/api/resource-set.js]: Missing `set` parameter everywhere, no ?

Comment (julien-f) [inline src/api/resource-set.js]: Indeed :p

File: src/api/resource-set.js (hunk 1 above)
Comment (julien-f) [inline src/api/resource-set.js]: Sounds good to me.

File: src/api/resource-set.js (hunk 2 above)
Comment (julien-f) [inline src/api/resource-set.js]: Maybe keeping these simple methods for manual use and have a `set({ id, name, objects, subjects })` better suited for your needs?

File: src/api/resource-set.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  return this.deleteResourceSet(id)
+}
+export { delete_ as delete }
+
+delete_.permission = 'admin'
+
+// -------------------------------------------------------------------
+
+export function get ({ id }) {
+  return id == null
```
Comment (marsaud) [inline src/api/resource-set.js]: A getAll would be nice too.

**Pull request merged**